### ISSUE TITLE
Improve docs: They are super confusing/hard to follow/convulated, in particular the k3s example

### ISSUE THREAD

**Issue by desmap** [containous/traefik#6106]
Cannot get k3s + traefik to work, tried a stripped down version of https://docs.traefik.io/v2.0/user-guides/crd-acme/#cluster-resources without LE:
1. Installed k3s with their script which installs traefik already (please update your docs according to the new k3s install script or offer a second version); it's also not clear what the k3s with helm installed version of traefik config is, hence update your example to the default install way
2. I applied the IngressResource def from your example
3. I applied the example manifests below

My browser just shows a 404 and IDK where to look and the whole thing slowly gets infuriating.
Since using k8s has much longer feedback loops, working with such huge examples frustrates. It would be awesome to have some really short example for k3s/k8s. Just plain yaml, just the most basic stuff.

Any help is highly appreciated. Thanks!

```
kind: Deployment
apiVersion: apps/v1
metadata:
  namespace: default
  name: whoami
  labels:
    app: whoami

spec:
  replicas: 2
  selector:
    matchLabels:
      app: whoami
  template:
    metadata:
      labels:
        app: whoami
    spec:
      containers:
        - name: whoami
          image: containous/whoami
          ports:
            - name: web
              containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: whoami

spec:
  ports:
    - protocol: TCP
      name: web
      port: 80
  selector:
    app: whoami

---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: simpleingressroute
  namespace: default
spec:
  entryPoints:
    - web
  routes:
  - match: Host(`my.domain.com`)
    kind: Rule
    services:
    - name: whoami
      port: 80
```

**Comment by desmap** [comment]
One more confusing thing: I just saw that the aforesaid page is just in your 2.x docs but pls note that the latest version of k3s doesn't ship with traefik 2.x yet (just check their used helm chart).

**Comment by desmap** [comment]
Got it finally working. Code is below and here's the reason: The given example on https://docs.traefik.io/v2.0/user-guides/crd-acme/#cluster-resources is confusing because of multiple reasons:

- It uses an old k3s image which is installed via docker BUT without its default traefik 1.7 (`--no-deploy traefik` is set)
- Then, the example should have installed any traefik version 2.x (because the example uses the CRD IngressRoute but not Ingress; I assume b/c traefik 2.x only works with IngressRoute)
- But then the example doesn't install anything, no traefik, no nothing, so you have an old k3s 0.8 without traefik and then a sea of yaml... (the CRD)

Most readers would assume to install k3s the standard way with the script (b/c then you get uninstall scripts etc.) or using k3d if you want to run k3s in a docker BUT NOT pulling yourself the image in some custom docker and without traefik. It's just noise when a one-liner could have achieved the same. Super weird. If there's a reasoning behind this I am super happy to learn why.

So, either I got all wrong or the example should install k3s via script w/o built-in traefik 1.7 and then install traefik 2.x OR just use k3d (also without traefik 1.7 and then get traefik 2.0). The only positive, they finally gave up on TOML.

for now, I went the k3s via script install with built-in traefik 1.7, b/c I just don't have time:
(besides it's not clear to me if they still support normal Ingress in 2.x or they just broke compatibility, maybe somebody knows)
```
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: default
  name: whoami
  labels:
    app: whoami
spec:
  replicas: 2
  selector:
    matchLabels:
      app: whoami
  template:
    metadata:
      labels:
        app: whoami
    spec:
      containers:
      - name: whoami
        image: containous/whoami
        ports:
        - name: web
          containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: whoami
spec:
  ports:
    - protocol: TCP
      name: web
      port: 80
  selector:
    app: whoami
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: ingress-whoami
  namespace: default
  annotations:
    kubernetes.io/ingress.class: traefik
spec:
  rules:
  - http:
      paths:
      - path: /
        backend:
          serviceName: whoami
          servicePort: web
```

**Comment by desmap** [comment]
So, I tried to get into the 2.x example but the port-forwarding part at https://docs.traefik.io/user-guides/crd-acme/#port-forwarding would need more explanations:

You write **And this will not work, because as it is, our Traefik pod is not reachable from the outside, which will make the ACME TLS challenge fail.** But why? To this point in the tutorial, everything should be set and once you apply the IngressRoutes, which is the next step, all shall be good. So, this needs further explanation.

Besides, this port-forwarding hack is not what a user expects from an proxy that handles LE fully automatic without any config, check also what another user wrote about this here: https://blog.dd19.de/blog/2019/11/15/traefik-2-0-on-microk8s/

It would be good to have asap some update on this 2.x guide/example.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

We dedicate the issue tracker to bug reports and feature requests only. My advanced AI has spotted that your issue might be a configuration problem or relates to something that doesn't look like a bug.

To confirm this, please join our [Community Forum](https://community.containo.us/) and reach out to us on the [Traefik section](https://community.containo.us/c/traefik).

In case I'm wrong (well, that would be embarrassing :sweat_smile:), my developers will re-open the issue and fix me!

In the meantime, you can double check [Traefik's documentation](https://docs.traefik.io/).
